service/src/user/UserService.ts

```diff
@@ -1 +1,42 @@
-export class UserService {}
+import {UserRepository} from "./UserRepository";
+import {User} from "./User";
+import {UserEmailAlreadyExistsError} from "./UserEmailAlreadyExistsError";
+import {UserRepeatedPasswordWrongError} from "./UserRepeatedPasswordWrongError";
+import {UserPasswordHasher} from "./UserPasswordHasher";
+import {UserIdGenerator} from "./UserIdGenerator";
+
+export class UserService {
+    constructor(
+        private userIdGenerator: UserIdGenerator,
+        private userRepository: UserRepository,
+        private userPasswordHasher: UserPasswordHasher,
+    ) {
+    }
+
+    async register(name: string, email: string, password: string, repeatedPassword: string): Promise<User>
+    {
+        if (await this.isEmailExists(email)) {
+            throw new UserEmailAlreadyExistsError(email);
+        }
+
+        if (!this.isCorrectRepeatedPassword(password, repeatedPassword)) {
+            throw new UserRepeatedPasswordWrongError();
+        }
+
+        //TODO: sendEmailConfirmation();
+
+        const hashedPassword = this.userPasswordHasher.hash(password);
+        const userId = this.userIdGenerator.generateUUID();
+
+        return await this.userRepository.create(new User(userId, name, hashedPassword, email));
```

Comment:
Await in case is redundant

```suggestion
        return this.userRepository.create(new User(userId, name, hashedPassword, email));
```

Reply:
ok, but not clear why.

Reply:
await we need if we want to get a value in the function or to `catch` some errors. But here we can just return Promise and "await" will be using in the calling place